cmd/experimental/kjobctl/apis/v1alpha1/application_profile_types.go

```diff
@@ -0,0 +1,106 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// VolumeBundleReference is the name of the VolumeBundle.
+// +kubebuilder:validation:MaxLength=253
+// +kubebuilder:validation:Pattern="^[a-z0-9]([-a-z0-9]*[a-z0-9])?(\\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*$"
+type VolumeBundleReference string
+
+type ApplicationProfileMode string
+
+const (
+	InteractiveMode ApplicationProfileMode = "Interactive"
+	JobMode         ApplicationProfileMode = "Job"
+)
+
+type Flag string
+
+const (
+	CmdFlag         Flag = "cmd"
+	ParallelismFlag Flag = "parallelism"
+	CompletionsFlag Flag = "completions"
+	RequestFlag     Flag = "request"
+	LocalQueueFlag  Flag = "localqueue"
+)
+
+type SupportedMode struct {
+	// name determines which template will be used and which object will eventually be created.
+	// Possible values are Interactive and Job.
+	//
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Enum=Job;Interactive
+	Name ApplicationProfileMode `json:"name"`
+
+	// requiredFlags point which cli flags are required to be passed in order to fill the gaps in the templates.
+	//
+	// +optional
+	// +kubebuilder:validation:Enum=cmd;parallelism;completions;request;localqueue
+	RequiredFlags []Flag `json:"requiredFlags"`
+
+	// interactive is a template (with defaults set) for running a Pod application.
+	//
+	// +optional
+	Interactive *corev1.PodSpec `json:"interactive,omitempty"`
+
+	// job is a template (with defaults set) for running a Job application.
+	//
+	// +optional
+	Job *batchv1.JobSpec `json:"job,omitempty"`
```

Comment:
We should be able to use CEL to validate that one and only one of the two fields is populated